docs/user/datastores/index_config.rst

```diff
@@ -260,6 +260,117 @@ If you are using the GeoMesa ``SftBuilder``, you may call the overloaded attribu
+
+Specifying a Table Splitter
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A table splitter may be specified by through user data when creating a simple feature type, before calling
+``createSchema``.
+
+To indicate the table splitter class, use the key ``table.splitter.class``:
+
+.. code-block:: java
+
+    sft.getUserData().put("table.splitter.class", "org.example.CustomSplitter");
+
+To indicate any options for the given table splitter, use the key ``table.splitter.options``:
+
+.. code-block:: java
+
+    sft.getUserData().put("table.splitter.options", "foo,bar,baz");
+
+See :ref:`set_sft_options` for details on setting user data.
+
+The Default Table Splitter
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Generally, ``table.splitter.class`` can be omitted. If so, GeoMesa will use a default implementation that allows
+for a flexible configuration using ``table.splitter.options``. If no options are specified, then all tables
+will generally create 4 split (based on the number of shards). The default ID index splits assume
+that feature IDs are randomly distributed UUIDs.
+
+For the default splitter, ``table.splitter.options`` should consist of comma-separated entries, in the form
+``key1:value1,key2:value2``. Entries related to a given index should start with the index identifier, e.g. one
+of ``id``, ``z3``, ``z2`` or ``attr`` (``xz3`` and ``xz2`` indices use ``z3`` and ``z2``, respectively).
+
++------------+-------------------------------+----------------------------------------+
+| Index Type | Option                        | Description                            |
++============+===============================+========================================+
+| Z3/XZ3     | ``z3.min``                    | The minimum date for the data          |
++            +-------------------------------+----------------------------------------+
+|            | ``z3.max``                    | The maximum date for the data          |
++            +-------------------------------+----------------------------------------+
+|            | ``z3.bits``                   | The number of leading bits to split on |
++------------+-------------------------------+----------------------------------------+
+| Z2/XZ2     | ``z2.bits``                   | The number of leading bits to split on |
++------------+-------------------------------+----------------------------------------+
+| ID         | ``id.pattern``                | Split pattern                          |
++------------+-------------------------------+----------------------------------------+
+| Attribute  |  ``attr.<attribute>.pattern`` | Split pattern for a given attribute    |
++------------+-------------------------------+----------------------------------------+
+
+Z3/XZ3 Splits
++++++++++++++
+
+Dates are used to split based on the Z3 time prefix (typically weeks). They are specified in the form
+``YYYY-MM-dd``. If the minimum date is specified, but the maximum date is not, it will default to the current date.
```

Comment:
should this now change to lower case yyyy ?

Reply:
yeah, for consistency, good catch

Reply:
fixed - I also updated a few places in the docs that I missed with the java time PR